storage/campaign.go

```diff
@@ -126,3 +126,19 @@ func (db *store) GetTotalComplaints(campaignID, userID int64) (int64, error) {
 	err := db.Table("complaints").Select("count(recipient)").Count(&totalComplaints).Where("campaign_id = ? and user_id= ?", campaignID, userID).Error
 	return totalComplaints, err
 }
+
+// GetCampaignBounces fetches campaign bounces by campaign id, and populates the pagination obj
+func (db *store) GetCampaignBounces(campaignID, userID int64, p *PaginationCursor) error {
+	p.SetCollection(&[]entities.Bounce{})
+	p.SetResource("bounces")
+
+	query := db.Table(p.Resource).
+		Where("campaign_id = ? and user_id=?", campaignID, userID).
+		Order("created_at desc, id desc").
+		Limit(p.PerPage)
+
+	p.SetQuery(query)
+
+	return db.Paginate(p, campaignID)
```

Comment:
Paginate takes a `userID` not a `campaignID`. Please review the other pagination methods as well for this.